data/eigerfan/src/EigerFan.cpp

```diff
@@ -373,6 +374,9 @@ void EigerFan::HandleStreamMessage(zmq::message_t &message, boost::shared_ptr<zm
         configuredOffset = 0;
         lastFrameSent = 0;
         num_frames_sent = 0;
+        for(int j=0; j<num_frames_consumed.size(); j++) {
+          num_frames_consumed[j] = 0;
+        }
```

Comment:
At this point I think num_frames_consumed[j] might be undefined?

From the docs:
Unlike std::map::operator[], this operator never inserts a new element into the container. Accessing a nonexistent element through this operator is undefined behavior.

Reply:
I had wondered if it wasn't defined - but convinced myself it was defined because I've defined it when the consumers are setup (see your comment above)? But perhaps that's not the case?

Reply:
So the vector is certainly defined, and when you push_back a 0 you define num_frames_consumed[0].  But to define 4 elements you would actually need to call push_back four times. You could use assign(n, 0); where n is the number of elements.

I've just realised that actually line 377 will only loop once because the size is 1 at that point.  So my original statement is wrong, it will not be undefined here, but I don't see how you are setting the number of elements to the number of consumers.

I think at line 377 you want something like
num_frames_consumed.assign(<number_of_consumers>, 0);
and then you could eliminate line 122 and the rest of 377-379.

Have you been able to test this code out?

Reply:
Hi Sky, sorry I now see why it works.  Stupid me for only looking at the code snippet and not the whole file.  You are pushing back 0 for every consumer, not in the constructor so it will work fine.

Reply:
So ignore my suggested change because what you have done already works.

Reply:
The code has been tested and it's even in the integration testing :)